### ISSUE TITLE
Some jerk stole this and threw it on the Unity Store

### ISSUE THREAD

**Issue by KellanHiggins** [KellanHiggins/UnityFBXExporter#23]
EDIT: For those coming from the review on the Unity Asset store, that asset did not license the use of this project correctly and from the tone on the review makes it seem like we are buddy buddy. This is not true.

There are differences between his project and mine, but he took all the hard work I put in along with a few other contributors, packed it into a DLL and released it on the asset store. So you can read my response (which sometimes gets angry) below and see for yourself if I am being unreasonable.

Also you are welcome to check out my source code and improve upon it here!

----------------------------------

Hey everyone,

Guess what? This is apparently good enough to be stolen (aka violate the MIT terms) and thrown up on the Unity Asset store for $18!

It is called FBX Exporter MA for Unity, by Madman Alliance and it is located at https://www.assetstore.unity3d.com/en/#!/content/73212.

They took my free package, compiled it into a DLL, removed any trace of the MIT license that I had attached (and had to be attached always as is tradition for an MIT license) and are making cash.

I've emailed Unity, but honestly I am just posting this so that if someone is frustrated with that package they can google it and end up here. And then see all the source code.

Thanks to @mikelortega for emailing me about this.

Also, is anyone interesting in helping me manage pull requests. I keep forgetting to test and update the repo?

Thanks for your guys' editions! They are great.

-Kellan

**Comment by KellanHiggins** [comment]
Oh and it looks like they are also stealing all your good works too. A note mentioned that vertex colour export (which @Totchinuko submitted) is getting pulled into their DLL as well.

I guess they only good thing about that is they are actually updating the repo :P. I need to get my act together.

**Comment by Totchinuko** [comment]
damn... But I'm not really a license professional... What is forbidden with MIT? I never quit understood how it work ._. (funny stuff, Those 3 commits were my first commit ever on git hub. Not sure how I should feel about this)

**Comment by KellanHiggins** [comment]
So the MIT license allows anyone to take and build upon the work, including just straight ripping it off and charging for it online. HOWEVER, you have to include the license that appears at the top of the source code or else you violate the terms of it. Which this guy did.

Also if everyone can go and upvote the review that mikelo left that'll put it at the top of the reviews.

@Totchinuko You were the second person I pulled code in from. People are going to steal, but making their lives difficult is all part of the game and you helped a ton of people out. One guy was literally emailing me about vertex colours just before you posted it. Thanks for that!

**Comment by Totchinuko** [comment]
Already upvoted that. And thanks for the little explanation. Cause with my project going on, I have quit some code under various MIT license. I wonder, how does it work if a piece of MIT code end up into a game DLL that you sell? Do you provide an MIT license file into your game files?

Your project helped me a lot to build my vertex painter and road generator so the resulting mesh don't end up cluttering the scene file (I still have problems with normal in exported files tho, but that's another story). I still have to test the exports now that I upgraded my project to unity 5.5... Will probably commit changes to my fork if I find any problem with that.

**Comment by KellanHiggins** [comment]
Hmm, yes you make a good argument. I think the crux here is that the person is literally just ripping us off and not displaying the license. It is going to be less enforceable in a big game compiled with lots of different things. You are find in the situation that you describe there. It would still be good to include that notice somewhere in your code though that anyone can see. The copyright notice has to be displayed according to the license. I don't think a lot of big games do this. I wouldn't worry about it as long as your game isn't only a single MIT license project.

Honestly if they just openly posted the copyright notice on the asset it would be fine with a link to our project. But then they would get zero sales as they haven't added anything to the package. They probably stole a binary converter from somewhere else.

Someone could take this project and upload it to the Asset Store tomorrow and as long as the copyright notice is posted, it is fine. They could have just uploaded it as source code and there would be no problem. Hiding it behind a DLL is super sketchy.

**Comment by Totchinuko** [comment]
Well, hopefully it will resolve nicely. I'll keep an eye on this ^^.

**Comment by madmanalliance** [comment]
Hi KellanHiggins,
    First of all, that is true that FBX Exporter MA for Unity has included some code from your Asset as a startup in mid 2016 because find Blender only support FBX binary for import. Lucky Blender has a great documentation to talk about the FBX binary specification(https://code.blender.org/2013/08/fbx-binary-file-format-specification/) and our works are base on it. We have do many rewriting to made it work to export binary format, expect one class - FBXUnityMaterialGetter and it is the reason mikelortega said FBX Exporter MA for Unity is your project. But in term of number of classes, FBX Exporter MA for Unity has at least 13 classes more to made it unique to it parent. Therefore, FBX Exporter MA for Unity is only a descendants of your project. We are not just copying from yours. We decide to sell it $18 is only for the cost of the binary format.

    For the MIT license problem, we have the same questions with Totchinuko that "how does it work if a piece of MIT code end up into a DLL that you sell" around the deployment and that why nothing in the package,  however we have keep your license header in that piece source code from beginning.  Base on your request here we are willing to credit back your project on the page. 

Hi Totchinuko,
   For the vertex colours, It seem that we doing it on the same time. There isn't a piece code take from yours for our implementation. Not same as yours,  we use MappingInformationType: "ByVertice" and ReferenceInformationType: "Direct" for our implementation.

**Comment by madmanalliance** [comment]
@KellanHiggins 
To let you know, the rewritten level, enclosed is a screen capture for your reference. 

![snap](https://cloud.githubusercontent.com/assets/26950933/24744603/317ce51a-1ae5-11e7-8009-018541bda1ce.png)

**Comment by KellanHiggins** [comment]
Thanks for getting in touch @madmanalliance. You will need to include the MIT license in your project readme and description or else this is a violation of Unity's terms (which I have read in full). Had you included this in your project description on the asset store initially, we would not be going down this route. Also you'd have a lot fewer sales is my guess.

Also I want to point out you are plagiarizing this project's description.

This is the readme for this project.

![plagerize2](https://cloud.githubusercontent.com/assets/5897704/24762545/9078aadc-1aa3-11e7-893b-eeee73b12e8f.JPG)

This is your readme (Taken on April 6). No 1 is almost the same as my description. (2) is word for word without any changes. (3) is a shortened from my description.

![plagerize](https://cloud.githubusercontent.com/assets/5897704/24762550/9342e750-1aa3-11e7-9603-125e245c1958.JPG)

So in conclusion, fuck you. Your asset is just a straight rip off, disguised by a DLL. I wouldn't be surprised if you ripped that binary converter off from somewhere else.

You want to prove you've added works to it? Post your source code somewhere and we can compare.

**Comment by madmanalliance** [comment]
Hey @KellanHiggins ,
I come here is to solve the conflict and not for argue. If I really a bad guy, I won't come here and discuss with you. I can simple shutdown the current asset and re-submit a new one by changing a little thing. Right? So please clam down and try to solve our conflict together.

OK! As per your request, I post FBXExporter.cs and  FBXUnityMaterialGetter.cs into a text file for you to compare and that is all I can disclose to public right now(As I working on Bone now, FBXUnityMeshGetter.cs for sorry).  

The main different are I have rewritten all stringbuffer into a class structure and made nodes for FBX 7.4. I have reused all your code that do work on material and Unity Asset database with minor changes. So what do you thing?
 
[source.txt](https://github.com/KellanHiggins/UnityFBXExporter/files/904565/source.txt)

Secondly, I will upload a new package to unity include your MIT license files in shortly. 

Thirdly, there hasn't an ascii to binary converter expect the Autodesk one. however there really has some writer exists, but they are mostly written in python, like https://github.com/ideasman42/pyfbx_i42 and and Blender's io_scene_fbx addon. Those provide a lot of help to me to writing the in fbx binary. Thats why my asset is the first FBX exporter that can write binary format without using Autodesk's fbx sdk.

**Comment by KellanHiggins** [comment]
You are coming off very reasonable, but frankly you have violated an MIT license and should no longer be able to use the works of this project.

The clear plagiarism is a very big deal. Copying the words from the readme is a big deal and can't be brushed away.

I am going to let Unity resolve this as they have gotten back to me. We will see what they say.

**Comment by KellanHiggins** [comment]
Welp, it looks like the asset was taken down by either the store or by the creator. I am going to go ahead and close this thread.

I would be happy to accept the code for the Binary conversion into this repo if they so decides to do that. A fair contribution considering they probably made some money off the asset. And then those who bought it from the asset store won't be unfairly disadvantaged.
